core/store/store_test.go

```diff
@@ -170,3 +176,121 @@ func TestStore_SyncDiskKeyStoreToDB_DBKeyAlreadyExists(t *testing.T) {
 	require.Len(t, keys, 1)
 	require.Equal(t, acc.Address.Hex(), keys[0].Address.String())
 }
+
+func TestStore_DeleteKey(t *testing.T) {
+	app, cleanup := cltest.NewApplicationWithKey(t,
+		cltest.EthMockRegisterGetBalance,
+	)
+	defer cleanup()
+	app.EthMock.Context("app.Start()", func(meth *cltest.EthMock) {
+		meth.Register("eth_chainId", app.Store.Config.ChainID())
+	})
+	require.NoError(t, app.StartAndConnect())
+	store := app.GetStore()
+
+	keys, err := store.AllKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 1)
+
+	err = store.DeleteKey(keys[0].Address.Address())
+	require.NoError(t, err)
+
+	keys, err = store.AllKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 0)
+}
+
+func TestStore_ArchiveKey(t *testing.T) {
+	app, cleanup := cltest.NewApplicationWithKey(t,
+		cltest.EthMockRegisterGetBalance,
+	)
+	defer cleanup()
+	app.EthMock.Context("app.Start()", func(meth *cltest.EthMock) {
+		meth.Register("eth_chainId", app.Store.Config.ChainID())
+	})
+	require.NoError(t, app.StartAndConnect())
+	store := app.GetStore()
+
+	var addrs []struct {
+		Address   common.Address
+		DeletedAt time.Time
+	}
+	err := store.DB.Raw(`SELECT address, deleted_at FROM keys`).Scan(&addrs).Error
+	require.NoError(t, err)
+
+	keys, err := store.AllKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 1)
+
+	err = store.ArchiveKey(keys[0].Address.Address())
+	require.NoError(t, err)
+
+	err = store.DB.Raw(`SELECT address, deleted_at FROM keys`).Scan(&addrs).Error
+	require.NoError(t, err)
+	require.Len(t, addrs, 1)
+
+	keys, err = store.SendKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 0)
+
+	keys, err = store.AllKeys()
+	require.NoError(t, err)
+	require.Len(t, keys, 0)
+}
+
+func TestKeyStore_Import(t *testing.T) {
```

Comment:
nitpick - some of these tests are named `TestStore_<Action>Key` and some are names `TestKeyStore_<Action>`